patches/api/0194-Add-Player-Client-Options-API.patch

```diff
@@ -118,15 +120,24 @@ index 0000000000000000000000000000000000000000..f7f171c4ee0b8339b2f8fbe82442d65f
 +    private final boolean chatColors;
 +    private final SkinParts skinparts;
 +    private final MainHand mainHand;
++    private final boolean allowServerListings;
++    private final boolean textFilteringEnabled;
 +
++    @Deprecated
 +    public PlayerClientOptionsChangeEvent(@NotNull Player player, @NotNull String locale, int viewDistance, @NotNull ChatVisibility chatVisibility, boolean chatColors, @NotNull SkinParts skinParts, @NotNull MainHand mainHand) {
++        this(player, locale, viewDistance, chatVisibility, chatColors, skinParts, mainHand, false, false);
++    }
++
++    public PlayerClientOptionsChangeEvent(@NotNull Player player, @NotNull String locale, int viewDistance, @NotNull ChatVisibility chatVisibility, boolean chatColors, @NotNull SkinParts skinParts, @NotNull MainHand mainHand, boolean allowServerListings, boolean textFilteringEnabled) {
```

Comment:
I feel like this ctor should take some other form, a set of wrapper objects that then get spread back out into individual fields otherwise deprecated ctors are just gonna stack up